File: python/leetcode/007.py
class Solution:
    def reverse(self, x):
        sign = 1 if x > 0 else -1
        valOfReverse, x = 0, abs(x)
        while x:
            valOfReverse = 10 * valOfReverse + x % 10
            x /= 10
        return sign * valOfReverse if -2**31 < valOfReverse < 2**31 - 1 else 0

File: cpp/leetcode/007.cpp
class Solution {
public:
    int reverse(int x) {
        long valOfReverse = 0;
        while (x) {
            valOfReverse = 10 * valOfReverse + x % 10;
            x /= 10;
        }
        if (valOfReverse > INT_MAX || valOfReverse < INT_MIN) {
            return 0;
        }
        return (int) valOfReverse;
    }
};